prov/psm2/src/psmx2_av.c

```diff
@@ -465,14 +474,21 @@ static int psmx2_av_connect_trx_ctxt(struct psmx2_fid_av *av,
 
 		/*
 		 * make it synchronous for now to:
-		 * (1) ensure array "req->errors" is valid;
-		 * (2) to simplify the logic of generating the final
-		 *     completion event.
+		 * (1) ensure the array "req->errors" is valid;
+		 * (2) simplify the logic of generating the final completion.
 		 */
 
 		if (req) {
+			/*
+			 * release the lock before calling progress function
+			 * because the address vector may be accessed when
+			 * processing polled completions.
+			 */
+			psmx2_unlock(&av->lock, 1);
```

Comment:
This temporary unlocking is not safe. An fi_av_insert() call could sneak in and change the av layout. Working on an alternative that doesn't need this unlocking.